cpp/src/slice2java/JavaUtil.cpp

```diff
@@ -1043,14 +1043,11 @@ Slice::JavaGenerator::getOptionalFormat(const TypePtr& type)
             {
                 return prefix + "VSize";
             }
-            case Builtin::KindObject:
-            {
-                return prefix + "Class";
-            }
             case Builtin::KindObjectProxy:
             {
                 return prefix + "FSize";
             }
+            case Builtin::KindObject:
             case Builtin::KindValue:
             {
```

Comment:
Should we add `assert(false)` here?

Reply:
We could, but I was going to hold off on this and write a blog to discuss where we want to draw the line.

The wrapper languages also have their own `optionalFormat` functions (what this function is) where we could also add `assert(false)` or throw:
https://github.com/zeroc-ice/ice/blob/4675c80836d70753d9e00db6583492c6f8969130/python/modules/IcePy/Types.cpp#L3046-L3049 And #2262 shows that even after all our other removal, there were some still some places where we writing optional classes by hand. Maybe we should add a check to `writeOptional` which throws if `format == OptionalFormat::Class`?

It's just not clear to me how far we should go here, so I was going to ask everyone else about it via blog.